docs/migration-guide.md

```diff
@@ -58,171 +82,63 @@ To add Storybook to a project that isn’t currently using Storybook:
-
-## Optional migrations
-
-In addition to the automigrations and manual migrations above, there are also optional migrations that you should consider. These are things that we’ve deprecated in Storybook 7 (but remain backwards compatible), or best practices that should help you be more productive in the future.
-
-These include:
-
-- Upgrade to Component Story Format v3 (CSF3), the latest iteration of our example format
-- Upgrade from the legacy storiesOf API to Component Story Format
-- Upgrade from `.stories.mdx` files to pure MDX and Component Story Format
-- Upgrade from Storybook 6 recipe “CSF stories with arbitrary MDX”
-
-### CSF2 to CSF3
-
-We published a [detailed post about CSF3](https://storybook.js.org/blog/storybook-csf3-is-here/) including all of its benefits and migration steps.
-
-If you want to just skip to the migration, we provide a codemod for your convenience which should automatically make the code changes for you (make sure to update the glob to fit your files):
+Then, you can optionally convert CSF 2 to CSF 3 ([why should you do this?](https://storybook.js.org/blog/storybook-csf3-is-here/)):
 
 ```sh
-npx storybook@latest migrate csf-2-to-3 --glob="src/**/*.stories.js"
+# Convert CSF 2 to CSF 3
+npx storybook@latest migrate csf-2-to-3 --glob="**/*.stories.tsx" --parser=tsx
 ```
 
-### storiesOf to CSF
-
-Storybook 7's architecture is focused on performance and needs code that is statically analyzable. For that reason, it does not work with `storiesOf`. We provide a codemod which, in most cases, should automatically make the code changes for you (make sure to update the glob to fit your files):
+### `storiesOf` to dynamically created stories
 
-```sh
-npx storybook@latest migrate storiesof-to-csf --glob="src/**/*.stories.tsx"
-```
+If you are using `storiesOf` for its ability to dynamically create stories, you can build your own "storiesOf" implementation by leveraging the new [(experimental) indexer API](https://storybook.js.org/docs/api/main-config-indexers). A proof of concept of such an implementation can be seen in [this StackBlitz demo](https://stackblitz.com/edit/github-h2rgfk?file=README.md). See the demo's README.md for a deeper explanation of the implementation.
 
-### .stories.mdx to MDX+CSF
+### `*.stories.mdx` to MDX+CSF
 
-Storybook 7 provides a cleaner [docs](./writing-docs/index.md) that defines manual documentation in pure MDX and stories in CSF, rather than the previous `.stories.mdx` hybrid approach, which is now deprecated. You can automatically convert your files using the following codemod (make sure to update the glob to fit your files):
+Storybook now requires that MDX pages now reference stories written in CSF, rather than the previous .stories.mdx hybrid approach. You can automatically convert your files using the following codemod (make sure to update the glob to fit your files):
 
 ```sh
 npx storybook@latest migrate mdx-to-csf --glob "src/**/*.stories.mdx"
```

Comment:
One thing: when you're able, replace this with a code snippet aimed at every package manager. We have it, and it's called `storybook-migration-mdx-stories.npm.mdx` and include the other package manager variants. The same pattern should be applied to the examples here, as we should provide what's currently supported and not have users tied to a specific command, leaving up to them to use what they feel comfortable with.

Reply:
@jonniebigodes — There is no snippet with that file name. There's also no harm in only providing the `npx` version of these codemod examples, as anyone with `node` installed already has `npx` available and none of these codemods relate to the project's package manager.

Reply:
@kylegach we have it it's [here](https://github.com/storybookjs/storybook/blob/next/docs/snippets/common/storybook-migration-mdx-stories.npm.mdx), and the pnpm version is [here](https://github.com/storybookjs/storybook/blob/next/docs/snippets/common/storybook-migration-mdx-stories.pnpm.mdx) and the Yarn version is [here](https://github.com/storybookjs/storybook/blob/next/docs/snippets/common/storybook-migration-mdx-stories.yarn.mdx).

Reply:
Thanks. It must've been added recently and I hadn't pulled down the latest `next` into this branch. 🤦‍♂️